libraries/common/src/main/java/androidx/media3/common/MimeTypes.java

```diff
@@ -590,6 +590,10 @@ public static String getMimeTypeFromMp4ObjectType(int objectType) {
         return C.ENCODING_DTS;
       case MimeTypes.AUDIO_DTS_HD:
         return C.ENCODING_DTS_HD;
+      case MimeTypes.AUDIO_DTS_EXPRESS:
+        return C.ENCODING_DTS; // This should be ENCODING_DTS_HD when IC platforms support it.
+      case MimeTypes.AUDIO_DTS_X:
+        return C.ENCODING_DTS; // This should be ENCODING_DTS_UHD_P2 when IC platforms support it.
```

Comment:
This is similar to the above one but slightly different:

`AudioFormat.ENCODING_DTS_UHD_P2` is [in Android 14 Beta](https://developer.android.com/reference/android/media/AudioFormat#ENCODING_DTS_UHD_P2), so currently we cannot refer to it in the constant class `C`. Could you please add `ENCODING_DTS_UHD_P2` in the `C` class and refer to its constant value (`0x0000001e`), then we can change it to `AudioFormat.ENCODING_DTS_UHD_P2` once it is released.

Reply:
I think this could be a problem because creation of AudioTrack with 0x0000001e will fail on the TV platforms that we are testing. I think it is because these Android TVs are not Android 14. So for now can we stick to ENCODING_DTS?

Reply:
Good point! I think we should look into this carefully, again, this method is also called by offload logic.

When looking at [if offload is supported or not](https://github.com/androidx/media/blob/release/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/DefaultAudioSink.java#L1716). The platform gets the int value returned by this method when evaluating the offload support, it tries to [convert the value](https://android.googlesource.com/platform/frameworks/base.git/+/refs/heads/android13-release/core/jni/android_media_AudioSystem.cpp#2300) to the native constant. If we pass an `ENCODING_DTS_UHD_P2` value, for Android <= 13, in the absence of definition of `ENCODING_DTS_UHD_P2`, the native constant converted to will be `AUDIO_FORMAT_INVALID`. Then I supposed that will lead to a "offload unsupported" result, then the DTS:X P2 stream cannot be offloaded played, when the device support `ENCODING_DTS` for offload playback.

cc @christosts - If we return `ENCODING_DTS` for mime type DTS:X P2, the offload logic will check against the support of `ENCODING_DTS` from the platform, if the platform said `ENCODING_DTS` is supported, an `AudioTrack` is created with `ENCODING_DTS`, then is it ok that DTS_UHD_P2 is actually played on that `AudioTrack`?

Reply:
Similar to previous comments, I think it's better to have logic to handle MimeTypes.AUDIO_DTS_X as ENCODING_DTS in components closer to the audio rather than putting it in MimeTypes, because putting this here means we map MimeTypes.AUDIO_DTS_X to ENCODING_DTS which is not accurate. 

> I think this could be a problem because creation of AudioTrack with 0x0000001e will fail on the TV platforms that we are testing. 

For ExoPlayer to create an AudioTrack with 0x0000001e it means the platform reported that such an encoding is supported in the first place, which should not happen.

Reply:
Please see 61a6b995104a3e40a9e27d223eceb8e79452ae2d.  C.ENCODING_DTS_UHD_P2 == 0x1e has been created.